src/Symfony/Component/ObjectMapper/ConditionCallableInterface.php

```diff
@@ -23,8 +23,9 @@
 interface ConditionCallableInterface
 {
     /**
-     * @param mixed $value  The value being mapped
-     * @param T     $object The object we're working on
+     * @param mixed  $value  The value being mapped
+     * @param T      $source The object we're working on
+     * @param T|null $target The target object to which the value will be mapped
      */
-    public function __invoke(mixed $value, object $object): bool;
+    public function __invoke(mixed $value, object $source, ?object $target): bool;
```

Comment:
I'm not sure about this yet, I still need to think about it. Know that right now this still works but if the property `foo` is present on both `B` and `C` the ObjectMapper will take the first target found.